src/client/pythonEnvironments/base/locators/lowLevel/fsWatchingLocator.ts

```diff
@@ -39,46 +40,60 @@ export abstract class FSWatchingLocator extends Locator {
         super();
     }
 
-    public async initialize(): Promise<void> {
-        if (this.initialized) {
-            return;
+    public async dispose(): Promise<void> {
+        if (this.disposables !== undefined) {
+            // tslint:disable-next-line:no-this-assignment
+            const { disposables } = this;
+            this.disposables = undefined;
+            await disposeAll(disposables);
         }
-        this.initialized = true;
-        this.startWatchers().ignoreErrors();
     }
 
-    public dispose(): void {
-        super.dispose();
-        this.initialized = false;
+    public async* iterEnvs(query?: PythonLocatorQuery): IPythonEnvsIterator {
+        if (this.disposables === undefined) {
+            this.disposables = await this.startWatchers();
```

Comment:
You do not need to await on `startWatchers()`.
```suggestion
            this.disposables = this.startWatchers().ignoreErrors();
```

Reply:
https://github.com/microsoft/vscode-python/pull/14713/files#r530688307